rust/lance-graph/src/parser.rs

```diff
@@ -1265,4 +1277,122 @@ mod tests {
+
+        match where_clause.expression {
+            BooleanExpression::ILike {
+                expression,
+                pattern,
+            } => {
+                match expression {
+                    ValueExpression::Property(prop) => {
+                        assert_eq!(prop.variable, "n");
+                        assert_eq!(prop.property, "name");
+                    }
+                    _ => panic!("Expected property expression"),
+                }
+                assert_eq!(pattern, "alice%");
+            }
+            _ => panic!("Expected ILIKE expression"),
+        }
+    }
+
+    #[test]
+    fn test_parse_ilike_case_insensitive_keyword() {
+        // Test that the ILIKE keyword itself is case-insensitive
+        let query = "MATCH (n:Person) WHERE n.name ilike 'ALICE%' RETURN n.name";
+        let result = parse_cypher_query(query);
+        assert!(result.is_ok(), "ilike (lowercase) should parse");
+
+        match &result.unwrap().where_clause.unwrap().expression {
+            BooleanExpression::ILike { pattern, .. } => {
+                assert_eq!(pattern, "ALICE%");
+            }
+            _ => panic!("Expected ILIKE expression"),
+        }
+    }
+
+    #[test]
+    fn test_parse_ilike_with_double_quotes() {
+        let query = r#"MATCH (n:Person) WHERE n.email ILIKE "%@EXAMPLE.COM" RETURN n.email"#;
+        let result = parse_cypher_query(query);
+        assert!(result.is_ok(), "ILIKE with double quotes should parse");
+
+        let ast = result.unwrap();
+        let where_clause = ast.where_clause.expect("Expected WHERE clause");
+
+        match where_clause.expression {
+            BooleanExpression::ILike { pattern, .. } => {
+                assert_eq!(pattern, "%@EXAMPLE.COM");
+            }
+            _ => panic!("Expected ILIKE expression"),
+        }
+    }
+
+    #[test]
+    fn test_parse_ilike_in_complex_where() {
+        let query = "MATCH (n:Person) WHERE n.age > 20 AND n.name ILIKE 'j%' RETURN n.name";
+        let result = parse_cypher_query(query);
+        assert!(result.is_ok(), "ILIKE in complex WHERE should parse");
+
+        let ast = result.unwrap();
+        let where_clause = ast.where_clause.expect("Expected WHERE clause");
+
+        match where_clause.expression {
+            BooleanExpression::And(left, right) => {
+                // Left should be age > 20
+                match *left {
+                    BooleanExpression::Comparison { .. } => {}
+                    _ => panic!("Expected comparison on left"),
+                }
+                // Right should be ILIKE
+                match *right {
+                    BooleanExpression::ILike { pattern, .. } => {
+                        assert_eq!(pattern, "j%");
+                    }
+                    _ => panic!("Expected ILIKE expression on right"),
+                }
+            }
+            _ => panic!("Expected AND expression"),
+        }
+    }
```

Comment:
Since `ilike` is highly similar to `like` and `like` has been tested in the codebase, maybe we can simply skip these three tests (they are very similar to the `like` tests)

Reply:
done